projects/media-viewer/src/lib/annotations/comment-set/comment-set-render.service.ts

```diff
@@ -50,9 +54,25 @@ export class CommentSetRenderService {
     }
   }
 
-  private height(element:any) {
+  private height(element: any) {
     return element.form.nativeElement.getBoundingClientRect().height;
   }
 
   private difference(a: number, b: number): number { return Math.abs(a - b); }
+
+  private makeSureWithinContainer(commentItem: CommentComponent, previousCommentItem: CommentComponent, containerHeight: number) {
+    containerHeight -= 10;
+    if (commentItem.commentBottomPos > containerHeight) {
+      commentItem.rectTop -= commentItem.commentBottomPos - containerHeight;
+    }
+    if (previousCommentItem) {
+      if (commentItem.commentBottomPos > previousCommentItem.rectTop) {
+        commentItem.rectTop -= commentItem.commentBottomPos - previousCommentItem.rectTop;
+      }
+    }
+    if (commentItem.rectTop < 0 ) {
+      commentItem.rectTop = 0;
```

Comment:
I guess we might end up with many comments set to rectTop=0, but there's nothing else we can do if there are too many

Reply:
correct